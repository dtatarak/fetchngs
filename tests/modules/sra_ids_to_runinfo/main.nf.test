nextflow_process {

    name "Test Process SRA_IDS_TO_RUNINFO"
    script "modules/local/sra_ids_to_runinfo.nf"
    process "SRA_IDS_TO_RUNINFO"

    test("Should run without failures without fields") {

        when {
            process {
                """
                input[0] = 'SRR9984183'
                input[1] = ''
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should run without failures with fields") {

        when {
            process {
                """
                input[0] = 'SRR9984183'
                input[1] = 'run_accession,experiment_accession,library_layout,fastq_ftp,fastq_md5'
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
