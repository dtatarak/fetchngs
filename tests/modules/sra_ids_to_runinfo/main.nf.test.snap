{
    "Should run without failures without fields": {
        "content": [
            {
                "0": [
                    "SRR9984183.runinfo.tsv:md5,6bc8bfb7058412e5a54b83e08d3d1673"
                ],
                "1": [
                    "versions.yml:md5,61ebf47bc00075513ff16295beddb197"
                ],
                "tsv": [
                    "SRR9984183.runinfo.tsv:md5,6bc8bfb7058412e5a54b83e08d3d1673"
                ],
                "versions": [
                    "versions.yml:md5,61ebf47bc00075513ff16295beddb197"
                ]
            }
        ],
        "timestamp": "2023-05-09T02:22:52+0000"
    },
    "Should run without failures with fields": {
        "content": [
            {
                "0": [
                    "SRR9984183.runinfo.tsv:md5,6419f18c29ab09634b3dd97a6b2acd06"
                ],
                "1": [
                    "versions.yml:md5,61ebf47bc00075513ff16295beddb197"
                ],
                "tsv": [
                    "SRR9984183.runinfo.tsv:md5,6419f18c29ab09634b3dd97a6b2acd06"
                ],
                "versions": [
                    "versions.yml:md5,61ebf47bc00075513ff16295beddb197"
                ]
            }
        ],
        "timestamp": "2023-05-09T02:22:52+0000"
    }
}